app/components/baseComponents/components/table/table.tsx

```diff
@@ -0,0 +1,33 @@
+import { TableData }  from "../../models/tableData";
+import styles from "./table.module.scss";
+export default function Table({tableData, tableClassName}:{ tableData: TableData, tableClassName?:String }){
+    let headerRow;
+    if(tableData.columnNames){
+        const headerColumnNames = tableData.columnNames.map(columnValue => <th className={styles.tableCell} scope="col"><strong>{columnValue}</strong></th>)
+        headerRow = <thead><tr className={styles.tableHeaderRow}>{headerColumnNames}</tr></thead>;
+    }
+    const getTableRow = (tableRow:any={})=> {
+        return <tr className={styles.tableRow}>
+            <th className={styles.tableCell}>{tableRow.rank}</th>
+            <td className={styles.tableCell}>{tableRow.name}</td>
+            <td className={styles.tableCell}>{tableRow.totalScore}</td>
+        </tr>
+    }
```

Comment:
Two small comments on this,
- Is this normal to just inline define a function like this? Does this need to be a function, not saying it shouldn't, but if assuming it should, why not create a function outside the actual component definition?
- It seems like the "assumption" that the first argument to this function will have a `rank`, a `name` and a `totalScore`, doesn't that mean that this entire component _(or at least this little)_ inner function is pretty specific to our current use of it and not to the more general idea that the name of the class implies, `Table`?

_Also, not blocking just questions that I think are worth asking._

Reply:
1. Yeah if the function is small enough, then yeah it's normal.
2. This is true, but I think for our case, that's the most likely use of tables. We can always iterate on this if we need to.